x/concentrated-liquidity/swaps_tick_cross_test.go

```diff
@@ -0,0 +1,741 @@
+package concentrated_liquidity_test
+
+import (
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/osmosis-labs/osmosis/v16/x/concentrated-liquidity/math"
+	"github.com/osmosis-labs/osmosis/v16/x/concentrated-liquidity/swapstrategy"
+	"github.com/osmosis-labs/osmosis/v16/x/concentrated-liquidity/types"
+)
+
+// SwapTickCrossTestSuite tests that the ticks are
+// updated correctly after swapping in various scenarios.
+type SwapTickCrossTestSuite struct {
+	KeeperTestSuite
+}
+
+func TestSwapTickCrossTestSuite(t *testing.T) {
+	suite.Run(t, new(SwapTickCrossTestSuite))
+}
+
+// CreatePositionTickSpacingsFromCurrentTick creates a position with the passed in tick spacings away from the current tick.
+func (s *SwapTickCrossTestSuite) CreatePositionTickSpacingsFromCurrentTick(poolId uint64, tickSpacingsAwayFromCurrentTick uint64) positionMeta {
+	pool, err := s.App.ConcentratedLiquidityKeeper.GetPoolById(s.Ctx, poolId)
+	s.Require().NoError(err)
+
+	currentTick := pool.GetCurrentTick()
+
+	lowerTickTwo := currentTick - int64(tickSpacingsAwayFromCurrentTick)*int64(pool.GetTickSpacing())
+	upperTickTwo := currentTick + int64(tickSpacingsAwayFromCurrentTick)*int64(pool.GetTickSpacing())
```

Comment:
any reason behind the name lowerTickTwo / upperTickTwo?

Also is it intentional that these ticks aren't rounded to the tick spacing?

Reply:
> any reason behind the name lowerTickTwo / upperTickTwo?

I missed changing this to a more general name after refactoring. Prior to refactoring, I was creating each position individually so named "Two" to reflect that this is for position "NR2"

Will fix the name

> Also is it intentional that these ticks aren't rounded to the tick spacing?

It ended up being rounded to the nearest tick spacing by construction since `currentTick` is always a multiple of 100 and 1 in this test setup.

I will make sure to add a guarantee that this works more generally (for any other tick spacing)

Reply:
addressed